letslambda.py

```diff
@@ -36,89 +31,75 @@
 # add ch to logger
 LOG.addHandler(handler)
 
-def load_config(bucket):
+def load_from_s3(conf, s3_key):
+    """
+    Try to load a file from the s3 bucket and return it as a string
+    Return None on error
+    """
+    try:
+        s3 = conf['s3_client']
+        content = s3.get_object(Bucket=conf['s3_bucket'], Key=s3_key)["Body"].read()
+    except ClientError as e:
+        LOG.error("Failed to load '{0}' in bucket '{1}'".format(s3_key, conf['s3_bucket']))
+        LOG.error("Error: {0}".format(e))
+        return None
+
+    return content
+
+def load_config(s3, s3_bucket, letslambda_config):
     """
     Try to load the letlambda.yml out of the user bucket
     Will return None if the configuration file does not exist
     """
-    conf = bucket.get_key("letslambda.yml")
-    if conf == None:
+
+    try:
+        conf = s3.get_object(Bucket=s3_bucket, Key=letslambda_config)["Body"].read()
```

Comment:
If I understand well, `boto3` offers a class to manipulate s3 objects in a more natual way : 
http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Object.get

Maybe we could use that ?